contributors/guide/style-guide.md

```diff
@@ -389,6 +392,10 @@ external appearance.
   - **Good example:** A _CustomResourceDefinition_ (CRD) extends the Kubernetes
     API.
   - **Bad example:** A CRD extends the Kubernetes API.
+- When referring to a Kubernetes Group (SIG, WG, or UG) do not use the hyphenated
+  form unless it is for a specific purpose such as a file-name or URI.
+  - **Good example:** SIG Docs oversees the Kubernetes website.
```

Comment:
This brings things inline with how sig docs refers to groups within their documentation (SIG Docs vs SIG-Docs).
/cc @justaugustus
Following up from convo aaaagggggesss ago.